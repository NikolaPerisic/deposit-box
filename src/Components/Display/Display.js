import React from "react";
import "./Display.scss";

const display = props => {
  return (
    <div
      className={
        props.backlight
          ? "panel__display"
          : "panel__display panel__display--off"
      }
    >
      <p>{props.isLocked ? "Locked" : "Unlocked"}</p>
      <h4>{props.message}</h4>
    </div>
  );
};

export default display;
