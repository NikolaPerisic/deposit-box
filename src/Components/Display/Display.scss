.panel__display {
  grid-area: a;
  display: grid;
  border-radius: 5px;
  color: #434343;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(3, 1fr);
  background-color: #7fffff;
  grid-template-areas:
    "c c c c"
    "d d d d"
    "e e e e";
  p {
    grid-area: c;
    padding: 0.5rem;
    margin: 0;
  }
  h4 {
    grid-area: e;
    text-align: end;
    padding: 0.5rem;
    margin: 0;
    font-size: 2rem;
  }
}
.panel__display--off {
  background-color: #47b2b2;
}
