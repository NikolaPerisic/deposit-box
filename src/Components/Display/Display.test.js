import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Display from "./Display";

describe("Display component", () => {
  //
  const props = {
    message: "Ready",
    isLocked: false,
    backlight: true
  };

  it("renders without errors", () => {
    const DisplayComponent = shallow(<Display {...props} />);
    expect(toJson(DisplayComponent)).toMatchSnapshot();
  });
  //
  it("renders panel display", () => {
    const DisplayComponent = shallow(<Display {...props} />);
    const wrapper = DisplayComponent.find(".panel__display");
    expect(wrapper.length).toBe(1);
  });
});
