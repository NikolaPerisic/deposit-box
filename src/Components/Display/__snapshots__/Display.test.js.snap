// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders without errors 1`] = `
<div
  className="panel__display"
>
  <span>
    Unlocked
  </span>
  <p>
    Ready
  </p>
</div>
`;
